mamba/utils.py

```diff
@@ -61,4 +52,36 @@ def to_package_record_from_subjson(subdir, pkg, jsn_string):
     info['channel'] = channel
     info['url'] = join_url(channel_url, pkg)
     package_record = PackageRecord(**info)
-    return package_record
\ No newline at end of file
+    return package_record
+
+def _make_virtual_package(name, version=None):
+    return PackageRecord(
+            package_type=PackageType.VIRTUAL_SYSTEM,
+            name=name,
+            version=version or '0',
+            build='0',
+            channel='@',
+            subdir=context.subdir,
+            md5="12345678901234567890123456789012",
+            build_number=0,
+            fn=name,
+    )
+
+def _supplement_index_with_system(index):
```

Comment:
Should the one from conda be used here? Otherwise you'll have to change this function every time `conda` adds a virtual package.

Reply:
In mamba `index` is a list, while it is a dict in `conda`, but yes we could transform it before/after the call.

Reply:
We now use conda's function (see #104).

Reply:
Thanks @davidbrochart